p2p/host/basic/basic_host.go

```
@@ -95,7 +97,11 @@ func (h *BasicHost) newConnHandler(c inet.Conn) {
 func (h *BasicHost) newStreamHandler(s inet.Stream) {
 	protoID, handle, err := h.Mux().ReadHeader(s)
 	if err != nil {
-		log.Error("protocol mux failed: %s", err)
+		if err == io.EOF {
+			log.Warningf("protocol EOF: %s", s.Conn().RemotePeer())
+		} else {
+			log.Errorf("protocol mux failed: %s", err)
```

I don't like to bury/ignore an error we have never seen.